### PR TITLE
fix: show error message when full local storage

### PR DESCRIPTION
**Pull request by nphi**

### CODE REVIEW

File: src/adapters/github.js [hunk 1]
```diff
@@ -342,16 +347,22 @@ class GitHub extends PjaxAdapter {
...
-              const oldestRepo = repos.reduce((min, p) => (hugeRepos[p] < hugeRepos[min] ? p : min));
-              delete hugeRepos[oldestRepo];
-            }
-            hugeRepos[repo] = new Date().getTime();
-            this.store.set(STORE.HUGE_REPOS, hugeRepos);
+
+          const handleHugeRepo = (isFullStorage) => this._handleError(cfg, {status: isFullStorage ? 600 : 206}, cb);
+
+          // The hugeRepos isn't a default {} due to the full Local Storage
+          if (!hugeRepos) return handleHugeRepo(true);
```
Comment (buunguyen) [inline src/adapters/github.js]: Why should store.get() return wrong data when storage is full? Are you sure about this?

File: src/adapters/github.js [hunk 2]
```diff
@@ -342,16 +347,22 @@ class GitHub extends PjaxAdapter {
...
-          if (!hugeRepos[repo]) {
-            // If there are too many repos memoized, delete the oldest one
-            if (repos.length >= GH_MAX_HUGE_REPOS_SIZE) {
-              const oldestRepo = repos.reduce((min, p) => (hugeRepos[p] < hugeRepos[min] ? p : min));
-              delete hugeRepos[oldestRepo];
-            }
-            hugeRepos[repo] = new Date().getTime();
-            this.store.set(STORE.HUGE_REPOS, hugeRepos);
+
+          const handleHugeRepo = (isFullStorage) => this._handleError(cfg, {status: isFullStorage ? 600 : 206}, cb);
```
Comment (buunguyen) [inline src/adapters/github.js]: Why 'handleHugeRepo' calls handleError? Just doesn't make any sense.

File: src/adapters/adapter.js [hunk 3]
```diff
@@ -197,7 +197,11 @@ class Adapter {
         }
 
         break;
+      case 600:
```
Comment (buunguyen) [inline src/adapters/adapter.js]: Isn't 600 still valid HTTP code? Should be something that outside any range.

File: src/adapters/github.js (hunk 1 above)
Comment (nphi) [reply src/adapters/github.js]: When the new user installs or update the Octotree v3 but his local storage is already full, the default setting is unable to set storage, which leads to the variable is undefined

**Comment by buunguyen** [comment]
Actually if there's no bug in storage code. I don't think we should do too much. We already log the error.

File: src/adapters/github.js (hunk 1 above)
Comment (buunguyen) [reply src/adapters/github.js]: In this case, should we just short circuit the code to return early, skipping the check huge repo logic? That would make things simple without a big and ugly change like this (now if it's less ugly, I'm okay, but this is really ugly).

File: src/adapters/github.js (hunk 2 above)
Comment (nphi) [reply src/adapters/github.js]: I will think a better name. Basically, the function just determines whether it should pass the full local storage status or the huge repo status to the `handleError`

**Pull request closed without merging**